src/lib/build.js

```diff
@@ -55,10 +66,52 @@ const runBuild = async (options) => {
       scheme: apiUrl.protocol.slice(0, -1),
       host: apiUrl.host,
     }
-    options = { ...options, testOpts }
+    buildOptions = { ...buildOptions, testOpts }
+  }
+
+  /** @type {Partial<import('../../utils/types').ServerSettings>} */
+  let settings = {}
+  try {
+    settings = await detectServerSettings(devConfig, commandOptions, site.root)
+
+    const defaultConfig = { build: {} }
+
+    if (settings.buildCommand) {
+      buildOptions.cachedConfig.config.build.command = settings.buildCommand
```

Comment:
This is the sort of question that is tricky to answer without types in place, but is there a chance that some of these intermediate properties will be `undefined`, leaving to a reference error?

Reply:
@eduardoboucas 

Did some chasing here, and found that cachedConfig is created by `@netlify/config` - and cachedconfig.config.build is always added. https://github.com/netlify/build/blob/d2c74c02280bd2324cb9b933df4be9ba8c399789/packages/config/src/base.js#L32

I'm not sure what I should do to safeguard against that situation popping up if there's a breaking change in `@netlify/config` though. The node version doesn't support optional chaining, and I don't want to do 4 different `hasOwnProperty`s :/ 

But wow, wouldn't this all be so much nicer with Typescript 😅